src/julia/tools.py

```diff
@@ -140,13 +140,31 @@ def julia_py_executable():
     """
     Path to ``julia-py`` executable installed for this Python executable.
     """
-    scripts_path = sysconfig.get_path("scripts")
-    stempath = os.path.join(scripts_path, "julia-py")
-    candidates = {os.path.basename(p): p for p in glob.glob(stempath + "*")}
+
+    # try to find installed julia-py script - check scripts folders under different installation schemes
+    # we check the alternate schemes first, at most one of which should give us a julia-py script
+    # see https://docs.python.org/3/install/index.html#alternate-installation
+    scripts_paths_failed = []
+    for scheme in sysconfig.get_scheme_names():
+        scripts_path = sysconfig.get_path("scripts", scheme)
+        stempath = os.path.join(scripts_path, "julia-py")
+        candidates = {os.path.basename(p): p for p in glob.glob(stempath + "*")}
+        if candidates:
+            break
+        else:
+            scripts_paths_failed.append(scripts_path)
+
+    # try standard install scripts path if nothing found in alternate scheme
+    if not candidates:
+        scripts_path = sysconfig.get_path("scripts")
+        stempath = os.path.join(scripts_path, "julia-py")
+        candidates = {os.path.basename(p): p for p in glob.glob(stempath + "*")}
+
     if not candidates:
+        scripts_paths_failed.append(scripts_path)
```

Comment:
It's tricky to figure out `scripts_paths_failed.append(scripts_path)` is not going to add the last `scripts_path` used in `for scheme in sysconfig.get_scheme_names()` (you have to track the state of `candidates`).  Let's rename it to `scripts_paths_tried` and unconditionally append to it in the `for` loop and the first `if not candidates` block.

Reply:
`scripts_paths_tried` is redundant with the suggestion to gather both alternate and standard paths together and then loop over `scripts_paths`. If we don't find any `candidates` then we've already tried all `scripts_paths` and we can report all `scripts_paths` as the ones we tried.